dev/io.openliberty.security.jakartasec.3.0.internal_fat/fat/src/io/openliberty/security/jakartasec/fat/tests/TokenValidationTests.java

```diff
@@ -0,0 +1,1219 @@
+    }
+
+    /**
+     * Build Subject token validation failure expectations.
+     *
+     * @return - subject specific token validation failure expectations
+     * @throws Exception
+     */
+    public Expectations getSubjectMismatchExpectations() throws Exception {
+
+        Expectations expectations = getBaseErrorExpectations();
+        if (_testName.contains("Missing")) {
+            // check for jose4j embedded message
+            expectations.addExpectation(new ServerMessageExpectation(rpServer, "No Subject", "Did not receive a message stating that the sub claim was missing."));
+        }
+        if (_testName.contains("Empty")) {
+            expectations.addExpectation(new ServerMessageExpectation(rpServer, MessageConstants.CWWKS2426E_TOKEN_HAS_EMPTY_CLAIM, "Did not find a message in the server log stating that sub claim was empty."));
+        }
+
+        return expectations;
+    }
+
+    /**
+     * Build Audience token validation failure expectations.
+     *
+     * @return - audience specific token validation failure expectations
+     * @throws Exception
+     */
+    public Expectations getAudiencetMismatchExpectations() throws Exception {
+
+        Expectations expectations = getBaseErrorExpectations();
+        expectations.addExpectation(new ServerMessageExpectation(rpServer, MessageConstants.CWWKS2424E_ISSUER_MISMATCH, "Did not find a message in the server log stating that the audience did not match the clientId."));
+
+        return expectations;
+    }
+
+    /**
+     * Build authorized party token validation failure expectations.
+     *
+     * @return - azp specific token validation failure expectations
+     * @throws Exception
+     */
+    public Expectations getAuthorizedPartyMismatchExpectations() throws Exception {
+
+        Expectations expectations = getBaseErrorExpectations();
+        expectations.addExpectation(new ServerMessageExpectation(rpServer, MessageConstants.CWWKS2424E_ISSUER_MISMATCH, "Did not find a message in the server log stating that the audience did not match the clientId."));
+
+        return expectations;
+    }
+
+    /**
+     * Build Expiration in the past token validation failure expectations.
+     *
+     * @return - exp specific token validation failure expectations
+     * @throws Exception
+     */
+    public Expectations getExpInPastExpectations() throws Exception {
+
+        Expectations expectations = getBaseErrorExpectations();
+        expectations.addExpectation(new ServerMessageExpectation(rpServer, MessageConstants.CWWKS2427E_TOKEN_IS_EXPIRED, "Did not find a message in the server log stating that the token has expired."));
+
+        return expectations;
+    }
+
+    /**
+     * Build Issued At in the future token validation failure expectations.
+     *
+     * @return - iat specific token validation failure expectations
+     * @throws Exception
+     */
+    public Expectations getIatInFutureExpectations(String claim) throws Exception {
```

Comment:
This method and the associated constant MessageConstants.CWWKS2428E_IAT_IN_FUTURE are misleading because the CWWKS2428E message was updated to refer to a generic claim value that's in the future; it's no longer a message specifically for the "iat" claim.

Reply:
Updated the method name to getTimeValueInFutureExpectations and the message variable name to CWWKS2428E_TIME_IN_FUTURE